oryx-common/src/protocols.rs

```diff
@@ -34,6 +37,17 @@ impl Display for TransportProtocol {
     }
 }
 
+impl FromStr for TransportProtocol {
+    type Err = ParseProtocolError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Udp" | "udp" => Ok(Self::UDP),
+            "Tcp" | "tcp" => Ok(Self::TCP),
+            _ => Err(ParseProtocolError),
+        }
+    }
+}
```

Comment:
You can consider using the `strum` crate for easier enum handling in Rust. See [`EnumString`](https://docs.rs/strum/latest/strum/derive.EnumString.html) and [`Display`](https://docs.rs/strum/latest/strum/derive.Display.html).

Reply:
Good one :+1: